### ISSUE TITLE
Memory leak ocrmypdf.ocr vs subprocess.run

### ISSUE THREAD

**Issue by CGarces** [ocrmypdf/OCRmyPDF#1046]
I'm using ocrmypdf 13.4.0+dfsg inside a docker container using ubuntu:22.04 as base image.

The process is inside a loop and handle several PDF files with this code.

```
    ocrmypdf.ocr(
        os.path.join(origin_valiable, "temp.pdf"),
        os.path.join(destination_valiable, "final.pdf"),
        tesseract_timeout=0,
        skip_text=True,
        output_type="pdfa-3",
        optimize=3,
    )
```
After each call the memory usage grows and not is released. I have used psutil to log the memory usage after and before the calls.

I have fixed the memory leak running ocrmypdf as subprocess
```
     subprocess.run(
         [
             "ocrmypdf",
             "--tesseract-timeout=0",
             "--skip-text",
             "--output-type",
             "pdfa-3",
             "-O 3",
        os.path.join(origin_valiable, "temp.pdf"),
        os.path.join(destination_valiable, "final.pdf")
         ]
     )
```
This is the correct way to handle memory issues on OCRmyPDF? There is a better way?

**Comment by jbarlow83** [comment]
There was a memory leak in pikepdf earlier this year. I assume the affected version is in ubuntu 22.04. If you upgrade both pikepdf and ocrmypdf to current versions that should resolve the issue.

**Comment by michaelbrant** [comment]
I'm using the jbarlow83/ocrmypdf Docker image (image ID 1f3cafdbc295). When I run ocrmypdf I'm seeing memory usage go up to 100% on my local machine and then the process crashes. Does the docker image need to be updated?

**Comment by jbarlow83** [comment]
@michaelbrant Please open a new issue with clear steps to reproduce and your example file or indicate one of the test/resources files that reproduces the issue.

**Comment by CGarces** [comment]
@jbarlow83 can you please reopen the issue?

Updating the ocrmypdf [as described in the documentation](https://ocrmypdf.readthedocs.io/en/latest/installation.html#installing-the-latest-version-on-ubuntu-22-04-lts) not fix the issue

```
sudo apt-get update
sudo apt-get -y install ocrmypdf python3-pip

pip install --user --upgrade ocrmypdf
```

This is part of my docker file using ubuntu:22.04

```
# Install dependencies
RUN apt-get update && apt-get upgrade -y && apt-get install  -y  --no-install-recommends \
    python3-pip\
    ocrmypdf tesseract-ocr-spa \
    libimage-exiftool-perl \
    curl

# Installs the Python requirements
RUN pip3 install --upgrade --no-cache-dir -r requirements.txt
```

Inside the docker:

```
# bash
root@97db5907374c:/app# ocrmypdf --version
14.0.1
root@97db5907374c:/app# pip show pikepdf
Name: pikepdf
Version: 5.0.1+dfsg
Summary: Read and write PDFs with Python, powered by qpdf
Home-page: https://github.com/pikepdf/pikepdf
Author: James R. Barlow
Author-email: james@purplerock.ca
License: MPL-2.0
Location: /usr/lib/python3/dist-packages
Requires:
Required-by: ocrmypdf
root@97db5907374c:/app#
```

So ocrmypdf has been updated to 14.0.1 but pikepdf remains at 5.0.1.

As commented on #992 looks like the memory leaks was solved on pikepdf  5.4

I'm not sure how the dependencies are build but, should be pyproject.toml dependencies upgraded?

https://github.com/ocrmypdf/OCRmyPDF/blob/master/pyproject.toml#L25

I have included pikepdf on requirements.txt yo force the update of pikepdf  to 6.x

The memory leak still happens so I think that this issue can be reopened.

**Comment by CGarces** [comment]
Upgrading all python packages make the situation worse.

Change the docker to made a "clean" instalation of ocrmypdf 14.0.1 also not solve the memory leak.

```
# Install dependencies
RUN apt-get update && apt-get upgrade -y && apt-get install  -y  --no-install-recommends \
    ghostscript \
    jbig2dec \
    img2pdf \
    libsm6 libxext6 libxrender-dev \
    pngquant \
    tesseract-ocr \
    tesseract-ocr-spa \
    unpaper \
    python3-pip\
    libimage-exiftool-perl \

RUN pip3 install --no-cache-dir -r requirements.txt

```
requirements.txt contains `ocrmypdf` as dependency

I'll keep 13.4.0 + subprocess.run
